### ISSUE TITLE
VERY cool tool, but has HIGH CPU usage

### ISSUE THREAD

**Issue by SirFrostingham** [eraser1/BE_AEG#1]
This tool runs at 25+% CPU usage.  If you introduce delay to the main loop, it would probably help.

Very cool tool though!  Thanks for building it.

---

EDIT: Just tested with introducing 1 second delay and it went from 25+% CPU usage down to 0% CPU usage.

code:

`
# include <chrono>
# include <thread>

//edit your scripts_exceptions routine...

scripts_exceptions:
    // performance gain: sleep 1 second to reduce CPU usage to 0%
    std::chrono::seconds duration(1);
    std::this_thread::sleep_for(duration);
`

**Comment by eraser1** [comment]
Thanks for the feedback!

I didn't catch your edit so I went googling and found "std::Sleep" (for windows) and "std::usleep" (for unix/linux), and it seems to work fine for me.

I also put the sleep stuff in a helper function, so I can call it around a bit more easily (and I did, since I want to sleep when I "goto" certain points).

Is there any reason/benefit to using "std::this_thread::sleep_for" instead of what I did? Perhaps better cross-platform support?

**Comment by SirFrostingham** [comment]
I fixed the code tags, so the code I sent will work. 

```
//start

//add these includes:
#include <chrono>
#include <thread>

//edit your scripts_exceptions routine...

scripts_exceptions:
// performance gain: sleep 1 second to reduce CPU usage to 0%
std::chrono::seconds duration(1);
std::this_thread::sleep_for(duration);

//end

```

I think your version would work too.  Just need to test on Windows.  :)  

RE: cross platform support - Yes. Certain methods result in cross platform from what I read.

Again, great tool man!  Great job!

**Comment by SirFrostingham** [comment]
I just tested and on Windows 10, it's still at 25+% CPU usage.  I wonder if you can get the code i sent to work?

**Comment by eraser1** [comment]
Just committed the fix.

I was accidentally calling `my_sleep` AFTER `goto`, so it would never actually sleep. Oops :p

Also, I tried your method, but it didn't support "sleep" for less than a second, so I used `std::chrono::milliseconds` instead. Then I did some digging and found that `std::this_thread::sleep_for` was, to some degree, a wrapper for WIN32 `Sleep`, so I went back to my method instead (I still left some preprocessor options if you want to try that out instead though).

I don't have any Linux system, nor do I have any plans for installing it, so I can't test my code in that environment, but you shouldn't be running ArmA on Linux anyways lol.

PS: I tried it on my machine (`Windows 7 Ultimate 64-bit` with an `i7-3770` quad-core with hyperthreading) and I only got up to 2% CPU usage (without sleeping).
